### ISSUE TITLE
JSON field is **always** dirty

### ISSUE THREAD

**Issue by browner12** [laravel/framework#30564]
### Description:

I'm not sure that this is something Laravel should fix, but thought I'd file it anyway as more of a PSA.  I've filed it to MySQL to see if there's anything that can be done there.

https://bugs.mysql.com/bug.php?id=97589&thanks=4

If you use a native MySQL 8 JSON field type and are casting that field to a Collection, it will **always** return the JSON with a space between each element.

For the most part, not an issue, but 1 place it can cause real issues is if you're checking if one of these fields is dirty.  MySQL will set the `original` field as the string *with* the spaces, but then if you set the field to an array of **the same array elements**, the internal `json_encode` of the Model will return a string *without* the spaces.

We could *possibly* handle it in the FW to have a special check for dirty JSON fields, but that seems like it might be a little messy.

This *may* work on MySQL 5 JSON fields and/or other JSON casted types, but this was the only scenario I checked on.

### Steps To Reproduce:

Create a table with a field that has a JSON field type.  Cast that field to a Collection.

```php
$user = User::find(1);
$user->emails = ['test@gmail.com', 'test2@gmail.com', 'test3@gmail.com'];

dd($user->isDirty('emails'));
```

**Comment by staudenmeir** [comment]
Rejected fix: #28029

**Comment by driesvints** [comment]
Closing this as the fix for this was rejected in the past.

Please also fill out the entire issue template.
